org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/TelemetryManager.java

```diff
@@ -106,9 +114,36 @@ public void onBuildFinished(long buildFinishedTime) {
 						sourceLevelMax = Float.parseFloat(sourceLevel);
 					}
 				}
+
+				try {
+					IMarker[] projectMarkers = project.findMarkers(null /* all markers */, true /* subtypes */, IResource.DEPTH_ZERO);
+					projectErrors += Stream.of(projectMarkers).filter(m -> m.getAttribute(IMarker.SEVERITY, 0) == IMarker.SEVERITY_ERROR).count();
+					IMarker[] allmarkers = project.findMarkers(null /* all markers */, true /* subtypes */, IResource.DEPTH_INFINITE);
+					unresolvedImportErrors += Stream.of(allmarkers).filter(m -> isUnresolvedImportError(m)).count();
+				} catch (CoreException e) {
+					// ignore
+				}
+
+				IJavaProject javaProject = ProjectUtils.getJavaProject(project);
+				if (javaProject != null) {
+					try {
+						IVMInstall vmInstall = JavaRuntime.getVMInstall(javaProject);
+						if (JavaRuntime.compareJavaVersions(vmInstall, sourceLevel) == -1) {
```

Comment:
What value will it return if for example: the `vm` is `17` and `sourceLevel` is `20`?

I'm trying to understand why only check `-1` but not all negatives.

Reply:
> What value will it return if for example: the vm is 17 and sourceLevel is 20?

It returns `-1`, because the number compare utility only return `-1`, `0`, `1`. See the code below.
```java
     public static int compare(long x, long y) {
        return (x < y) ? -1 : ((x == y) ? 0 : 1);
    }
```

Anyway, I will change it to `JavaRuntime.compareJavaVersions(vmInstall, sourceLevel) < 0` for better readability.